lib/publiccloud/eks.pm

```diff
@@ -24,26 +36,57 @@ Retrieves the full name of the uploaded image or die.
 =cut
 sub push_container_image {
     my ($self, $image, $tag) = @_;
+    $self->{tag} = $tag;
 
-    my $region     //= $self->region();
-    my $repository //= get_var("PUBLIC_CLOUD_CONTAINER_IMAGES_REPO", 'suse-qec-testing');
+    my $region //= $self->region();
     my $aws_account_id   = $self->{aws_account_id};
     my $full_name_prefix = "$aws_account_id.dkr.ecr.$region.amazonaws.com";
-    my $full_name        = "$full_name_prefix/$repository:$tag";
-
-    $self->{repository} = $repository;
-    $self->{tag}        = $tag;
+    my $full_name        = "$full_name_prefix/$self->{repository}:$tag";
 
     assert_script_run("aws ecr get-login-password --region $region | docker login --username AWS --password-stdin $full_name_prefix");
     assert_script_run("docker tag $image $full_name");
     assert_script_run("docker push $full_name", 180);
     return $full_name;
 }
 
-sub cleanup {
+=head2 eks_apply_manifest
+Apply a kubernetes manifest 
+=cut
+sub eks_apply_manifest {
+    my ($self, $manifest) = @_;
+
+    my $env      = get_kubectl_env();
+    my $filename = create_manifest($manifest);
+
+    assert_script_run("$env kubectl apply -f $filename");
+}
+
+=head2 get_kubectl_env
+Get the environment variables needed to call kubectl
+=cut
+sub get_kubectl_env {
+    my $eks_key    = get_required_var('PUBLIC_CLOUD_EKS_KEY');
+    my $eks_secret = get_required_var('PUBLIC_CLOUD_EKS_SECRET');
+    'AWS_ACCESS_KEY_ID="' . $eks_key . '" AWS_SECRET_ACCESS_KEY="' . $eks_secret . '"';
+}
+
+=head2 get_container_image_name
+Get the full name for a container image
+=cut
+sub get_container_image_name {
+    my ($self, $tag) = @_;
+    my $region //= $self->region();
+    my $aws_account_id   = $self->{aws_account_id};
+    my $full_name_prefix = "$aws_account_id.dkr.ecr.$region.amazonaws.com";
+    "$full_name_prefix/$self->{repository}:$tag";
+}
+
+=head2 clean_image
+Delete a ECR image
+=cut
+sub clean_image {
```

Comment:
I would name it `delete_image` or `remove_image` as you define above, or how the command sounds `batch-delete-image`